ui/src/components/pod_logs_viewer/hooks/useLogsEmitter.js

```diff
@@ -0,0 +1,58 @@
+import { useCallback, useEffect, useState } from "react";
+import { appConfig } from "../../../config";
+import { useTuringPollingApiEmitter } from "../../../hooks/useTuringPollingApiEmitter";
+import { useLogsApiEmitter } from "./useLogsApiEmitter";
+
+export const useLogsEmitter = (
+  apiEndpoint,
+  query,
+  extractTimestamp,
+  processLogs
+) => {
+  const { podLogs: configOptions } = appConfig;
+  const [apiOptions, setApiOptions] = useState({ query });
+
+  const setQuery = useCallback(
+    (setQuery) => {
+      setApiOptions((options) => ({
+        ...options,
+        query: setQuery(options.query),
+      }));
+    },
+    [setApiOptions]
+  );
+
+  useEffect(() => {
+    setQuery(() => query);
+  }, [query, setQuery]);
+
+  const { emitter: apiEmitter } = useTuringPollingApiEmitter(
+    apiEndpoint,
+    apiOptions,
+    configOptions.pollInterval
+  );
+
+  useEffect(() => {
+    apiEmitter.on("data", (entries) => {
+      const lastTimestamp = extractTimestamp(entries);
+
+      if (!!lastTimestamp) {
+        setQuery((q) => ({
+          ...q,
+          since_time: lastTimestamp,
+          head_lines: configOptions.batchSize,
```

Comment:
CMIIW, but per our [Swagger docs](https://github.com/gojek/turing/blob/main/api/api/specs/logs.yaml#L54) for the Pod logs API, `If both tail_lines and head_lines are set, tail_lines will be ignored`. Soo it seems to me like the `head_lines` should not be set if the user chose an option besides "from the container start".

Reply:
This is expected. When user chose for example "Last 100 records", then first API call is
```
?taill_ines=100
``` 
Then, we fetch a timestamp of the latest log entry from the first call AND (just in case) limit the max number of records (i.e. `head_lines`) that will be fetched in the next (and subsequent) API calls:
```
?head_lines=500&since_time=2021-09-09:12:00:00.234Z&tail_lines=ignored
```